middleware/schema/schema.go

```diff
@@ -170,6 +170,18 @@ type Metadata struct {
 	PluginDownloadURL string
 }
 
+func (m *Metadata) PopulatePackageSpec(pkg *schema.PackageSpec) {
+	pkg.DisplayName = m.DisplayName
+	pkg.Description = m.Description
+	pkg.Keywords = m.Keywords
+	pkg.Homepage = m.Homepage
+	pkg.Repository = m.Repository
+	pkg.Publisher = m.Publisher
+	pkg.LogoURL = m.LogoURL
+	pkg.License = m.License
+	pkg.PluginDownloadURL = m.PluginDownloadURL
+}
+
```

Comment:
This change feels unnecessary. I'll leave a comment explaining how we can get around it in `provider.go`.